internal/daemon/controller/handlers/scopes/scope_service_test.go

```diff
@@ -1727,3 +1727,137 @@ func TestListKeys(t *testing.T) {
 		})
 	}
 }
+
+func TestRotateKeys(t *testing.T) {
+	tc := controller.NewTestController(t, nil)
+	t.Cleanup(tc.Shutdown)
+
+	aToken := tc.Token()
+	uToken := tc.UnprivilegedToken()
+
+	iamRepoFn := func() (*iam.Repository, error) {
+		return tc.IamRepo(), nil
+	}
+	serversRepoFn := func() (*server.Repository, error) {
+		return tc.ServersRepo(), nil
+	}
+	authTokenRepoFn := func() (*authtoken.Repository, error) {
+		return tc.AuthTokenRepo(), nil
+	}
+
+	privCtx := auth.NewVerifierContext(
+		context.Background(),
+		iamRepoFn,
+		authTokenRepoFn,
+		serversRepoFn,
+		tc.Kms(),
+		&authpb.RequestInfo{
+			PublicId:       aToken.Id,
+			EncryptedToken: strings.Split(aToken.Token, "_")[2],
+			TokenFormat:    uint32(auth.AuthTokenTypeBearer),
+		})
+
+	unprivCtx := auth.NewVerifierContext(
+		context.Background(),
+		iamRepoFn,
+		authTokenRepoFn,
+		serversRepoFn,
+		tc.Kms(),
+		&authpb.RequestInfo{
+			PublicId:       uToken.Id,
+			EncryptedToken: strings.Split(uToken.Token, "_")[2],
+			TokenFormat:    uint32(auth.AuthTokenTypeBearer),
+		})
+
+	org, proj := iam.TestScopes(t, tc.IamRepo())
+
+	org.Name = "defaultOrg"
+	org.Description = "defaultOrg"
+	org, _, err := tc.IamRepo().UpdateScope(context.Background(), org, 1, []string{"Name", "Description"})
+	require.NoError(t, err)
+
+	// Add new role for listing+rotating keys in org
+	rotateKeysRole := iam.TestRole(t, tc.DbConn(), org.PublicId)
+	_, err = tc.IamRepo().AddRoleGrants(context.Background(), rotateKeysRole.PublicId, 1, []string{"id=*;type=*;actions=rotate-keys,list-keys"})
+	require.NoError(t, err)
```

Comment:
I know this is probably copied from the ListKeys tests, but could you look into using `iam.TestRoleGrant` instead to assign grants to roles? I think it should make this code a little easier to read, and I didn't know about it when I wrote the ListKeys tests.